src/libos/src/process/do_clone.rs

```diff
@@ -30,13 +30,16 @@ pub async fn do_clone(
     // TODO: add pointer checking
     let thread_entry = unsafe { *(user_rsp as *mut usize) };
 
+    // Set the default user fsbase to an address on user stack, which is
+    // a relatively safe address in case the user program uses %fs before
+    // initializing fs base address.
     let init_cpu_state = CpuContext {
         gp_regs: GpRegs {
             rsp: user_rsp as _,
             rip: thread_entry as _,
             ..Default::default()
         },
-        fs_base: new_tls.unwrap_or(0) as _,
+        fs_base: new_tls.unwrap_or(current!().vm().get_stack_limit()) as _,
```

Comment:
The stack information in vm is used by the first thread, If the current thread is not the first one, the implementation maybe wrong.